model/posts_model.js

```diff
@@ -1,111 +1,40 @@
-const config = require('../config')
+const config = require('../config');
 const db = require('pg-bricks').configure(config[process.env.NODE_ENV || 'development']);
 const Rx = require('rx');
 
-module.exports.addPosts = (description, url, tags, owner) => {
-    return Rx.Observable.create(observer => {
-        db.insert('posts', {description, 'image_url': url, 'user_id': owner}).returning('*').row((err, row) => {
-            if (err)
-                observer.onError(err);
-            else
-                observer.onNext(row);
-            observer.onCompleted();
-        });
-    });
-}
+module.exports.addPosts = (description, url, tags, owner) => db.insert('posts', {description, 'image_url': url, 'user_id': owner}).returning('*').row();
 
-module.exports.getPostByID = (id) => {
-    const postInfoObservable = Rx.Observable.create(observer => {
-        db.raw(`SELECT P.id as post_id, P.description, P.image_url FROM POSTS as P, USERS as U WHERE P.id = ${id.id} AND P.user_id = U.id`).row((err, row) => {
-            if (err)
-                observer.onError(err);
-            else
-                observer.onNext(row);
-            observer.onCompleted();
-        });
-    });
+module.exports.getPostByID = ({id}) => {
+    const getPost = db.select('p.id AS post_id', 'p.description', 'p.image_url').from('posts as p').join('users u').on('p.user_id', 'u.id').where({'p.id': id}).row();
 
-    const userInfoObservable = Rx.Observable.create(observer => {
-        db.raw(`SELECT U.id as user_id, U.username, U.profile_picture_url FROM POSTS as P, USERS as U WHERE P.id = ${id.id} AND P.user_id = U.id`).row((err, row) => {
-            if (err)
-                observer.onError(err);
-            else
-                observer.onNext(row);
-            observer.onCompleted();
-        });
-    });
+    const getUserInfo = db.select('u.id as user_id', 'u.profile_picture_url').from('users as u').join('posts p').on('u.id', 'p.user_id').where({'p.id': id}).row();
 
-    return Rx.Observable.forkJoin(postInfoObservable, userInfoObservable);
-}
+    return Promise.all([getPost, getUserInfo]);
+};
 
 // TODO: add sort
-module.exports.search = (queryParams) => {
-    const {tags, offset = 0, limit = 15, sort, description = ""} = queryParams;
-
+module.exports.search = async ({tags, offset = 0, limit = 15, sort, description = ""}) => {
     let findParams = {};
-    if(tags)
+    if (tags)
         findParams.tags = tags;
-    if(description)
-        findParams.description = {$regex: new RegExp(description, "i")};
+    if (description)
+        findParams.description = description;
 
-    const getPosts = (observer, postIDs) => {
+    const getPosts = async (postIDs) => {
         const hasIDs = postIDs !== undefined;
```

Comment:
would probably changed it to `const hasIds = !!postIDs` as "undefined" is tricky to work with (you can litteraly change the value of undefined lol), and because it's more robust to work with a falsy value (in case `postIDs` is **null** for example)

Reply:
Yeah I learned that about undefined a few weeks ago, really cool 
`!!undefined == false`

Reply:
TIL

Reply:
`!!anything` will give you the Boolean value of anything; `!anything` gives you the Boolean value as well, but the opposite value (truthy -> falsy, falsy -> truthy)